--- a/qtbase/src/corelib/global/qfloat16.h
+++ b/qtbase/src/corelib/global/qfloat16.h
@@ -45,6 +45,8 @@
 #include <QtCore/qmetatype.h>
 #include <string.h>
 
+#include <limits>
+
 #if defined(QT_COMPILER_SUPPORTS_F16C) && defined(__AVX2__) && !defined(__F16C__)
 // All processors that support AVX2 do support F16C too. That doesn't mean
 // we're allowed to use the intrinsics directly, so we'll do it only for
